compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2821,7 +2821,7 @@ public void visit(BLangLetExpression letExpression) {
         for (BLangLetVariable letVariable : letExpression.letVarDeclarations) {
             semanticAnalyzer.analyzeDef((BLangNode) letVariable.definitionNode, letExpression.env);
         }
-        BType exprType = checkExpr(letExpression.expr, letExpression.env);
+        BType exprType = checkExpr(letExpression.expr, letExpression.env, letExpression.expectedType);
```

Comment:
Is `letExpression.expectedType` here different from `this.expType`? If not, I think we should use `this.expType` instead because that may be what is set consistently. 

`expectedType` is set to the expression more for post type-checking analysis.

Reply:
Yes. Both are same here. Changed to use `this.expType` instead.